loki/transform/transform_array_indexing.py

```diff
@@ -33,9 +33,16 @@
 ]
 
 
-def shift_to_zero_indexing(routine):
+def shift_to_zero_indexing(routine, ignore=()):
     """
     Shift all array indices to adjust to 0-based indexing conventions (eg. for C or Python)
+
+    Parameters
+    ----------
+    routine : :any:`Subroutine`
+        The subroutine in which the array dimensions should be shifted
+    ignore : list of str
+        Dimensions (or rather variables being dimensions) to be ignored
```

Comment:
Maybe something like this would be more descriptive?
```suggestion
        List of variable names for which, if found in the dimension expression of an array subscript, that dimension is not shifted to zero.
```